lib/core_extensions/active_record/relation.rb

```diff
@@ -81,6 +81,24 @@ def window!(name, **opts)
         self
       end
 
+      # The LIMIT BY clause permit to improve deduplication based on a unique key, it has better performances than
+      # the GROUP BY clause
+      #
+      #   users = User.limit_by(1, id)
```

Comment:
`User.limit(1, by: :id)` or `User.limit(1).by(:id)` feel like more ruby-esque APIs, but I'm probably more tricky to implement?

Reply:
ah, based on further reading the last option is not a valid suggestion.